src/pdf/xmp.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @typedef {Object} Metadata
+ * @property {string} title
+ * @property {string} author
+ * @property {string} creator
+ * @property {Date} createdAt
+ */
+
+/**
+ * @param {import("pdfkit/js/pdfkit.standalone.js")} pdf
+ * @param {Metadata} metadata
+ */
+export function addXmp(pdf, metadata) {
+    var xmpRef = pdf.ref({ Type: "Metadata", Subtype: "XML" });
+    xmpRef.compress = false;
+    xmpRef.write(xmp(metadata));
+    xmpRef.end();
+    pdf._root.data.Metadata = xmpRef;
+}
+
+/**
+ * @param {Metadata} metadata
+ */
+export function xmp(metadata) {
+    var createdAt = metadata.createdAt.toISOString();
+    var timestamp = metadata.createdAt.getTime();
+    var uuid = timestamp + "-" + Math.floor(424242 * Math.random());
+
+    return [
```

Comment:
for readability is prefer template literals
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
Agreed, but this project intentionally tries to avoid syntax and features that are not supported by the targeted browsers. That includes template literals.